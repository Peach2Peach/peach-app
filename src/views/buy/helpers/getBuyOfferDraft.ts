type GetBuyOfferDraftParams = {
  minBuyAmount: number
  maxBuyAmount: number
  meansOfPayment?: MeansOfPayment
}
export const getBuyOfferDraft = ({
  minBuyAmount,
  maxBuyAmount,
  meansOfPayment = {},
}: GetBuyOfferDraftParams): BuyOfferDraft => ({
  type: 'bid',
  creationDate: new Date(),
  lastModified: new Date(),
  amount: [minBuyAmount, maxBuyAmount],
  meansOfPayment,
  paymentData: {},
  releaseAddress: '',
  originalPaymentData: [],
})
