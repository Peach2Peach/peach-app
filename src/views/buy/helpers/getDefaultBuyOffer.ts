type DefaultOfferProps = {
  minBuyAmount: number
  maxBuyAmount: number
  meansOfPayment?: MeansOfPayment
  kyc?: boolean
}
export const getDefaultBuyOffer = ({
  minBuyAmount,
  maxBuyAmount,
  meansOfPayment = {},
  kyc = false,
}: DefaultOfferProps): BuyOfferDraft => ({
  type: 'bid',
  creationDate: new Date(),
  lastModified: new Date(),
  amount: [minBuyAmount, maxBuyAmount],
  meansOfPayment,
  paymentData: {},
  releaseAddress: '',
  originalPaymentData: [],
  kyc,
})
