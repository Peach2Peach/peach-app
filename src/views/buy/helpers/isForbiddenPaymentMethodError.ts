import { isPaymentMethod } from '../../../utils/validation/isPaymentMethod'

export const isForbiddenPaymentMethodError = (
  errorMessage: string | null,
  errorDetails: unknown,
): errorDetails is PaymentMethod[] =>
  errorMessage === 'FORBIDDEN' && Array.isArray(errorDetails) && errorDetails.every(isPaymentMethod)
