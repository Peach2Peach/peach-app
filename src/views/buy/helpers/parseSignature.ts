const signaturePattern = /[a-zA-Z0-9/=+]{88}/u

export const parseSignature = (signature: string) => signature.match(signaturePattern)?.pop() || signature
