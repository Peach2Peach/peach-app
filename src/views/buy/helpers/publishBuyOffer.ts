import { publishPGPPublicKey } from '../../../init/publishPGPPublicKey'
import { saveOffer } from '../../../utils/offer'
import { postBuyOffer } from '../../../utils/peachAPI'

export const publishBuyOffer = async (offerDraft: BuyOfferDraft) => {
  let [result, err] = await postBuyOffer(offerDraft)

  if (err?.error === 'PGP_MISSING') {
    await publishPGPPublicKey()
    const response = await postBuyOffer(offerDraft)
    result = response[0]
    err = response[1]
  }
  if (result) {
    saveOffer({ ...offerDraft, ...result })
    return { offerId: result.id, isOfferPublished: true, errorMessage: null }
  }
  return { isOfferPublished: false, errorMessage: err?.error || 'POST_OFFER_ERROR' }
}
