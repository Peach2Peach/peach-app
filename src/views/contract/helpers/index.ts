export { isPSBTForBatch } from "./isPSBTForBatch";
export { shouldShowTradeStatusInfo } from "./shouldShowTradeStatusInfo";
export { verifyReleasePSBT as verifyPSBT } from "./verifyReleasePSBT";
