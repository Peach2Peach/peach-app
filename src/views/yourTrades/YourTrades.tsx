import { SectionList, View } from 'react-native'
import { Header, Loading, Screen } from '../../components'

import { createMaterialTopTabNavigator } from '@react-navigation/material-top-tabs'
import { useEffect } from 'react'
import { TabBar } from '../../components/ui/TabBar'
import { useNavigation, useRoute } from '../../hooks'
import { useTradeSummaries } from '../../hooks/query/useTradeSummaries'
import { useShowErrorBanner } from '../../hooks/useShowErrorBanner'
import tw from '../../styles/tailwind'
import i18n from '../../utils/i18n'
import { headerIcons } from '../../utils/layout'
import { parseError } from '../../utils/result'
import { SectionHeader } from './components/SectionHeader'
import { TradePlaceholders } from './components/TradePlaceholders'
import { TradeItem } from './components/tradeItem'
import { getPastOffers, isOpenOffer } from './utils'
import { getCategories } from './utils/getCategories'

const YourTradesTab = createMaterialTopTabNavigator()
const tabs = ['buy', 'sell', 'history'] as const

export const YourTrades = () => {
  const { tradeSummaries, isLoading, error, refetch } = useTradeSummaries()
  const showErrorBanner = useShowErrorBanner()
  useEffect(() => {
    if (error) showErrorBanner(parseError(error))
  }, [error, showErrorBanner])

  const { params } = useRoute<'yourTrades'>()
  const allOpenOffers = tradeSummaries.filter(({ tradeStatus }) => isOpenOffer(tradeStatus))
  const summaries = {
    buy: allOpenOffers.filter(({ type }) => type === 'bid'),
    sell: allOpenOffers.filter(({ type }) => type === 'ask'),
    history: getPastOffers(tradeSummaries),
  }

  return (
    <Screen style={tw`px-0`} header={<YourTradesHeader />} showFooter>
      <YourTradesTab.Navigator
        initialRouteName={params?.tab || 'buy'}
        tabBar={TabBar}
        sceneContainerStyle={[tw`px-sm`, tw.md`px-md`]}
      >
        {tabs.map((tab) => (
          <YourTradesTab.Screen
            key={tab}
            name={`yourTrades.${tab}`}
            children={() => (
              <>
                {summaries[tab].length > 0 ? (
                  <SectionList
                    contentContainerStyle={[tw`bg-transparent py-7`, isLoading && tw`opacity-60`]}
                    onRefresh={refetch}
                    refreshing={false}
                    showsVerticalScrollIndicator={false}
                    sections={getCategories(summaries[tab])}
                    renderSectionHeader={SectionHeader}
                    renderSectionFooter={() => <View style={tw`bg-transparent h-7`} />}
                    renderItem={TradeItem}
                    ItemSeparatorComponent={() => <View onStartShouldSetResponder={() => true} style={tw`h-6`} />}
                  />
                ) : (
                  <TradePlaceholders tab={tab} />
                )}
              </>
            )}
          />
        ))}
      </YourTradesTab.Navigator>
      {isLoading && (
        <View style={tw`absolute inset-0 items-center justify-center`} pointerEvents="none">
          <Loading />
        </View>
      )}
    </Screen>
  )
}

function YourTradesHeader () {
  const navigation = useNavigation()
  const onPress = () => {
    navigation.navigate('exportTradeHistory')
  }
  return (
    <Header
      title={i18n('yourTrades.title')}
      icons={[
        {
          ...headerIcons.share,
          onPress,
          accessibilityHint: `${i18n('goTo')} ${i18n('exportTradeHistory.title')}`,
        },
      ]}
      hideGoBackButton
    />
  )
}
