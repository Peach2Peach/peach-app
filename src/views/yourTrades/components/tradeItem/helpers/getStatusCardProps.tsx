import { Icon } from '../../../../../components'
import tw from '../../../../../styles/tailwind'
import { getThemeForTradeItem, isContractSummary, isPastOffer } from '../../../utils'
import { statusCardStyles } from '../../../../../components/statusCard/StatusCard'
import { contractIdToHex } from '../../../../../utils/contract'
import { getLevel, getAction } from '../utils'
import { offerIdToHex } from '../../../../../utils/offer'
import { getShortDateFormat } from '../../../../../utils/date/getShortDateFormat'

export const getStatusCardProps = (item: TradeSummary, navigateToOffer: () => void, navigateToContract: () => void) => {
  const tradeTheme = getThemeForTradeItem(item)
  const { tradeStatus, paymentMade, creationDate, id, amount, unreadMessages } = item
  const { price, currency } = isContractSummary(item) ? item : { price: undefined, currency: undefined }
  const status = tradeTheme.level === 'mild' ? 'waiting' : tradeStatus
  const title = isContractSummary(item) ? contractIdToHex(id) : offerIdToHex(id)
  const color = isContractSummary(item) ? getLevel(tradeTheme) : getLevel(tradeTheme, item)
  const date = isContractSummary(item) ? new Date(paymentMade || creationDate) : new Date(creationDate)
  const subtext = getShortDateFormat(date)
  const action = isContractSummary(item)
    ? getAction(item, navigateToContract, status)
    : getAction(item, navigateToOffer, tradeStatus)

  const icon = isPastOffer(item.tradeStatus) ? (
    <Icon id={tradeTheme.icon} size={16} color={statusCardStyles.border[color].borderColor} />
  ) : undefined

  return {
    title,
    icon,
    subtext,
    amount,
    price,
    currency,
    label: action.label,
    labelIcon: action?.icon && <Icon id={action.icon} size={16} color={statusCardStyles.text[color].color} />,
    onPress: action.callback,
    unreadMessages,
    color,
  }
}
