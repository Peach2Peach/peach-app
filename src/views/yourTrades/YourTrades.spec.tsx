import { fireEvent, render, waitFor } from '@testing-library/react-native'
import { Attributes, ReactElement, createElement } from 'react'
import { NavigationAndQueryClientWrapper } from '../../../tests/unit/helpers/NavigationAndQueryClientWrapper'
import { navigateMock } from '../../../tests/unit/helpers/NavigationWrapper'
import { queryClient } from '../../../tests/unit/helpers/QueryClientWrapper'
import { YourTrades } from './YourTrades'

jest.useFakeTimers()

jest.mock('../../hooks/useRoute', () => ({
  useRoute: jest.fn(() => ({
    params: {
      tab: 'buy',
    },
  })),
}))

jest.mock('@react-navigation/material-top-tabs', () => ({
  createMaterialTopTabNavigator: jest.fn(() => ({
    Navigator: (props: Attributes) => createElement('Navigator', props),
    Screen: (props: { children: () => ReactElement }) => createElement('Screen', props, props.children()),
  })),
}))

describe('YourTrades', () => {
  const wrapper = NavigationAndQueryClientWrapper
  it('should render correctly', async () => {
    const { toJSON } = render(<YourTrades />, { wrapper })
    await waitFor(() => {
      expect(queryClient.isFetching()).toBe(0)
    })
    expect(toJSON()).toMatchSnapshot()
  })

  it('should navigate to "exportTradeHistory" when clicking on the icon in the header', () => {
    const { getByAccessibilityHint } = render(<YourTrades />, { wrapper })
    const icon = getByAccessibilityHint('go to export trade history')
    fireEvent.press(icon)

    expect(navigateMock).toHaveBeenCalledWith('exportTradeHistory')
  })
})
