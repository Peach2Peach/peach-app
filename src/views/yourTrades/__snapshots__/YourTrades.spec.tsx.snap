// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourTrades should render correctly 1`] = `
<View
  style={
    {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      [
        {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
        },
        {
          "backgroundColor": "#FFF9F6",
        },
        {
          "borderColor": "#FEEDE5",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
            "paddingBottom": 6,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 6,
          },
          {
            "paddingLeft": 32,
            "paddingRight": 32,
          },
          false,
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexBasis": "0%",
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "gap": 4,
          }
        }
      >
        <Text
          allowFontScaling={false}
          numberOfLines={1}
          style={
            [
              {
                "color": "#2B1911",
                "fontFamily": "Baloo2-Regular",
                "fontSize": 16,
                "letterSpacing": 0.16,
                "lineHeight": 24,
              },
              [
                {
                  "fontFamily": "Baloo2-Bold",
                  "fontSize": 16,
                  "letterSpacing": 0.16,
                  "lineHeight": 26,
                },
                {
                  "fontFamily": "Baloo2-Bold",
                  "fontSize": 20,
                  "letterSpacing": 0.2,
                  "lineHeight": 32,
                },
                {
                  "color": "#2B1911",
                },
                {
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ],
              false,
            ]
          }
        >
          your trades
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 10,
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessibilityHint="go to export trade history"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
            }
          }
        >
          <test-file-stub
            fill="#9F8C82"
            style={
              [
                {
                  "height": 24,
                  "width": 24,
                },
                {
                  "height": 20,
                  "width": 20,
                },
                {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <RCTSafeAreaView
    style={
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "paddingBottom": 32,
            "paddingLeft": 32,
            "paddingRight": 32,
            "paddingTop": 32,
          },
          {
            "paddingLeft": 0,
            "paddingRight": 0,
          },
        ]
      }
    >
      <Navigator
        initialRouteName="buy"
        sceneContainerStyle={
          [
            {
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            {
              "paddingLeft": 32,
              "paddingRight": 32,
            },
          ]
        }
        tabBar={[Function]}
      >
        <Screen
          name="yourTrades.buy"
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
                "paddingLeft": 60,
                "paddingRight": 60,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                [
                  {
                    "color": "#2B1911",
                    "fontFamily": "Baloo2-Regular",
                    "fontSize": 16,
                    "letterSpacing": 0.16,
                    "lineHeight": 24,
                  },
                  {
                    "color": "#9F8C82",
                    "fontFamily": "Baloo2-Bold",
                    "fontSize": 20,
                    "letterSpacing": 0.2,
                    "lineHeight": 32,
                  },
                  false,
                ]
              }
            >
              none yet!
            </Text>
            <View
              style={
                [
                  {
                    "backgroundColor": "#EAE3DF",
                    "flexShrink": 1,
                    "height": 1,
                    "width": "100%",
                  },
                  {
                    "backgroundColor": "#F4EEEB",
                    "marginBottom": 32,
                    "marginTop": 32,
                    "width": "100%",
                  },
                ]
              }
            />
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  [
                    {
                      "color": "#2B1911",
                      "fontFamily": "Baloo2-Regular",
                      "fontSize": 16,
                      "letterSpacing": 0.16,
                      "lineHeight": 24,
                    },
                    {
                      "color": "#F56522",
                      "fontFamily": "Baloo2-Bold",
                      "fontSize": 20,
                      "letterSpacing": 0.2,
                      "lineHeight": 32,
                    },
                    false,
                  ]
                }
              >
                start buying!
              </Text>
              <test-file-stub
                fill="#F56522"
                style={
                  [
                    {
                      "height": 24,
                      "width": 24,
                    },
                    {
                      "height": 20,
                      "marginLeft": 8,
                      "width": 20,
                    },
                  ]
                }
              />
            </View>
          </View>
        </Screen>
        <Screen
          name="yourTrades.sell"
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
                "paddingLeft": 60,
                "paddingRight": 60,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                [
                  {
                    "color": "#2B1911",
                    "fontFamily": "Baloo2-Regular",
                    "fontSize": 16,
                    "letterSpacing": 0.16,
                    "lineHeight": 24,
                  },
                  {
                    "color": "#9F8C82",
                    "fontFamily": "Baloo2-Bold",
                    "fontSize": 20,
                    "letterSpacing": 0.2,
                    "lineHeight": 32,
                  },
                  false,
                ]
              }
            >
              none yet!
            </Text>
            <View
              style={
                [
                  {
                    "backgroundColor": "#EAE3DF",
                    "flexShrink": 1,
                    "height": 1,
                    "width": "100%",
                  },
                  {
                    "backgroundColor": "#F4EEEB",
                    "marginBottom": 32,
                    "marginTop": 32,
                    "width": "100%",
                  },
                ]
              }
            />
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  [
                    {
                      "color": "#2B1911",
                      "fontFamily": "Baloo2-Regular",
                      "fontSize": 16,
                      "letterSpacing": 0.16,
                      "lineHeight": 24,
                    },
                    {
                      "color": "#F56522",
                      "fontFamily": "Baloo2-Bold",
                      "fontSize": 20,
                      "letterSpacing": 0.2,
                      "lineHeight": 32,
                    },
                    false,
                  ]
                }
              >
                start selling!
              </Text>
              <test-file-stub
                fill="#F56522"
                style={
                  [
                    {
                      "height": 24,
                      "width": 24,
                    },
                    {
                      "height": 20,
                      "marginLeft": 8,
                      "width": 20,
                    },
                  ]
                }
              />
            </View>
          </View>
        </Screen>
        <Screen
          name="yourTrades.history"
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
                "paddingLeft": 60,
                "paddingRight": 60,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                [
                  {
                    "color": "#2B1911",
                    "fontFamily": "Baloo2-Regular",
                    "fontSize": 16,
                    "letterSpacing": 0.16,
                    "lineHeight": 24,
                  },
                  {
                    "color": "#9F8C82",
                    "fontFamily": "Baloo2-Bold",
                    "fontSize": 20,
                    "letterSpacing": 0.2,
                    "lineHeight": 32,
                  },
                  false,
                ]
              }
            >
              none yet!
            </Text>
          </View>
        </Screen>
      </Navigator>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#FFF9F6",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 8,
            "paddingTop": 8,
          },
          {
            "paddingBottom": 0,
            "paddingTop": 16,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
            "gap": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "height": 24,
              "width": 24,
            }
          }
        >
          <test-file-stub
            fill="#7D675E"
            style={
              [
                {
                  "height": 24,
                  "width": 24,
                },
                {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#2B1911",
                "fontFamily": "Baloo2-Regular",
                "fontSize": 16,
                "letterSpacing": 0.16,
                "lineHeight": 24,
              },
              [
                {
                  "color": "#7D675E",
                },
                {
                  "fontFamily": "Baloo2-SemiBold",
                  "fontSize": 9,
                  "letterSpacing": 0.16,
                  "lineHeight": 14.625,
                  "textAlign": "center",
                },
              ],
              false,
            ]
          }
        >
          buy btc
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
            "gap": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "height": 24,
              "width": 24,
            }
          }
        >
          <test-file-stub
            fill="#7D675E"
            style={
              [
                {
                  "height": 24,
                  "width": 24,
                },
                {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#2B1911",
                "fontFamily": "Baloo2-Regular",
                "fontSize": 16,
                "letterSpacing": 0.16,
                "lineHeight": 24,
              },
              [
                {
                  "color": "#7D675E",
                },
                {
                  "fontFamily": "Baloo2-SemiBold",
                  "fontSize": 9,
                  "letterSpacing": 0.16,
                  "lineHeight": 14.625,
                  "textAlign": "center",
                },
              ],
              false,
            ]
          }
        >
          sell btc
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
            "gap": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "height": 24,
              "width": 24,
            }
          }
        >
          <test-file-stub
            fill="#7D675E"
            style={
              [
                {
                  "height": 24,
                  "width": 24,
                },
                {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#2B1911",
                "fontFamily": "Baloo2-Regular",
                "fontSize": 16,
                "letterSpacing": 0.16,
                "lineHeight": 24,
              },
              [
                {
                  "color": "#7D675E",
                },
                {
                  "fontFamily": "Baloo2-SemiBold",
                  "fontSize": 9,
                  "letterSpacing": 0.16,
                  "lineHeight": 14.625,
                  "textAlign": "center",
                },
              ],
              false,
            ]
          }
        >
          wallet
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
            "gap": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "height": 24,
              "width": 24,
            }
          }
        >
          <test-file-stub
            fill="#7D675E"
            style={
              [
                {
                  "height": 24,
                  "width": 24,
                },
                {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#2B1911",
                "fontFamily": "Baloo2-Regular",
                "fontSize": 16,
                "letterSpacing": 0.16,
                "lineHeight": 24,
              },
              [
                {
                  "color": "#7D675E",
                },
                {
                  "fontFamily": "Baloo2-SemiBold",
                  "fontSize": 9,
                  "letterSpacing": 0.16,
                  "lineHeight": 14.625,
                  "textAlign": "center",
                },
              ],
              false,
            ]
          }
        >
          trades
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
            "gap": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "height": 24,
              "width": 24,
            }
          }
        >
          <test-file-stub
            style={
              {
                "height": 24,
                "width": 24,
              }
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "color": "#2B1911",
                "fontFamily": "Baloo2-Regular",
                "fontSize": 16,
                "letterSpacing": 0.16,
                "lineHeight": 24,
              },
              [
                {
                  "color": "#7D675E",
                },
                {
                  "fontFamily": "Baloo2-SemiBold",
                  "fontSize": 9,
                  "letterSpacing": 0.16,
                  "lineHeight": 14.625,
                  "textAlign": "center",
                },
              ],
              false,
            ]
          }
        >
          settings
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
